File: Backend/VechileAuctionProject/src/main/java/com/project/base/services/PaymentService.java
package com.project.base.services;

import com.project.base.pojo.Payment;
import com.project.base.pojo.PaymentFor;

public interface PaymentService {

    Payment createOrder(Double amount, Long userId,
                        PaymentFor paymentFor, Long referenceId) throws Exception;

    void verifyPayment(String orderId, String paymentId,
                       String signature, Long userId) throws Exception;
}

File: frontend/app/src/services/paymentService.js
import axios from "../services/axios";

export const createOrder = async ({ amount, paymentFor, referenceId }) => {
  const res = await axios.post("/user/payment/create-order", null, {
    params: { amount, paymentFor, referenceId }
  });
  return res.data;
};

export const verifyPayment = async (data) => {
  return axios.post("/user/payment/razorpay/verify", data);
};

